### PR TITLE
znapzend: 0.21.0 -> 0.23.2

### PR DESCRIPTION
**Pull request by Ma27**
## Description of changes
Changes:
* https://github.com/oetiker/znapzend/releases/tag/v0.21.1
* https://github.com/oetiker/znapzend/releases/tag/v0.22.0
* https://github.com/oetiker/znapzend/releases/tag/v0.23.0
* https://github.com/oetiker/znapzend/releases/tag/v0.23.1
* https://github.com/oetiker/znapzend/releases/tag/v0.23.2

Additionally a few minor things improved:

* Only use autoreconfHook, drop manual autotools invocations.
* Split outputs.
* Add very simple `installCheckPhase` to confirm that `znapzend` can resolve all of its Perl dependencies.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/tools/backup/znapzend/default.nix
Comment (smancill) [inline pkgs/tools/backup/znapzend/default.nix:39]: No need for this comment, actually. We prefer to use `refs/tags/<version>`.

File: pkgs/tools/backup/znapzend/default.nix
Comment (smancill) [inline pkgs/tools/backup/znapzend/default.nix:56]:
```suggestion
  installCheckPhase = ''
    runHook preInstallCheck
    $out/bin/znapzend --version
    runHook postInstallCheck
  '';
```

File: pkgs/tools/backup/znapzend/default.nix
Comment (smancill) [inline pkgs/tools/backup/znapzend/default.nix:61]:
```suggestion
  meta = {
    description = "High performance open source ZFS backup with mbuffer and ssh support";
    homepage    = "https://www.znapzend.org";
    license     = lib.licenses.gpl3;
```

File: pkgs/tools/backup/znapzend/default.nix
Comment (smancill) [inline pkgs/tools/backup/znapzend/default.nix:63]:
```suggestion
    maintainers = with lib.maintainers; [ otwieracz ma27 ];
    platforms   = lib.platforms.all;
```

**Review by smancill — DISMISSED** [review]
Please try to format with `nixfmt-rfc-style` also, so we fully modernize the derivation.

File: pkgs/tools/backup/znapzend/default.nix
Comment (Ma27) [reply pkgs/tools/backup/znapzend/default.nix:39]: No.
This was annoying to find out and I'll leave it as-is to not run into this again when I touch it next time.

**Comment by Ma27** [comment]
So, I fail to see much value in this review here: apart from the valid thing about `installCheckPhase` you're not only pointing out stuff that's completely orthogonal to this change. The main reason why this isn't merged is because I really want more testers here.

Complaints about formatting and this religious, unnuanced "`with` super-bad" are something that should be done by a linter in a CI pipeline _at best_. In fact, there's even a process in place for getting all of nixpkgs formatted gradually. I'm doing this for a few years now and I frequently observed that people have different opinions and will enforce different practices, at worst they even contradict. For me, it may be easy to counter that, but I'm convinced that this is a super frustrating experience for new committers if their changes are blocked on the opinions of maintainers.

If you care that deeply about this stuff, you're free to help out with getting RFC166 fully implemented. But pushing for _arbitrary_ things in individual PRs won't help much in the longrun.

**Comment by mweinelt** [comment]
@smancill That kind of attitude is not appreciated in the NixOS ecosystem.

**Comment by Ma27** [comment]
Got a second tester, so we should be good now.

**Pull request merged**
